Draft: Surfer community spotlight

TheZoq2 opened the pull request:
Here is a draft of my community spotlight post. I still need to do some small stuff like record a demo of the keyboard interface and perhaps of the Spade integration, but most of the content is here now.

CC: @oscargus, @LucasKl since they are involved in the project

mattvenn commented on content/post/2023/13-spotlight-Surfer.md:
```diff
@@ -0,0 +1,154 @@
+
+Apart from being a cool tech demo, this also enables some cool workflow in
+CI/CD. For example, we can <!-- have if we end up doing this before the blog
+release --> augment the tinytapeout test action to link to surfer with the
```
that would be cool! otherwise, just a link to tinytapeout

mattvenn reviewed (commented):
good draft, left a few comments

TheZoq2 commented on content/post/2023/13-spotlight-Surfer.md:
```diff
@@ -0,0 +1,154 @@
+
+Perhaps my favorite thing about Surfer right now is that it works almost
+flawlessly in web assembly. You can see a demo of that at
+[https://app.surfer-project.org](https://app.surfer-project.org), and if you
```
Default signals would be nice for sure, I'll try to come up with a clean way to do that.

What security warnings do you get?

mattvenn replied:
works now, weird. Maybe my connection was bad momentarily

TheZoq2 commented on content/post/2023/13-spotlight-Surfer.md:
```diff
@@ -0,0 +1,154 @@
+---
+title: "Community Spotlight - Surfer"
+date: 1970-01-01T13:26:31+01:00
```
I'll do that once this is finalized

TheZoq2 commented:
Pushed an update, all the TODOs I had are now fixed. I added some youtube embeds because I couldn't figure out how to embed the videos directly in hugo, hopefully that's fine :)

TheZoq2 commented on content/post/2023/13-spotlight-Surfer.md:
```diff
@@ -0,0 +1,164 @@
+flawlessly in web assembly. You can see a demo of that at
+[https://app.surfer-project.org](https://app.surfer-project.org). If you
+have a VCD file to analyse, you can just append that to the URL along with some commands to run when the waveform is loaded, for example
+<https://app.surfer-project.org/?load_url=https://app.surfer-project.org/picorv32.vcd&startup_commands=module_add%20testbench;divider_add%20.;divier_add%20top;module_add%20testbench.top>.
```
Oops, this link doesn't work, need to investigate

mattvenn commented on content/post/2023/13-spotlight-Surfer.md:
```diff
@@ -0,0 +1,167 @@
+Surfer is a new waveform viewer with a focus on extensibility and a snappy
+(optionally keyboard driven UI). It runs both natively and in web-assembly, so
+you can try it out right here in the browser at
+<https://app.surfer-project.org>.
```
maybe link this to something that opens a vcd?

mattvenn commented:
I was just thinking that first link is the one people are likely to click on, so it might be better to have it open a vcd file. Anyway, no strong feelings about it, up to you. Happy to merge as is.

TheZoq2 commented:
That's a valid point, I'll make it open a VCD file but not open any traces, because then it doesn't show the initial help message, I'll update that and fix the date, then I'm happy with this

TheZoq2 commented:
I just realized that it is thanksgiving over in the US today, so perhaps not the best time to release a blog post, perhaps we should hold it off until next week?

mattvenn commented:
no problem for me. I'll merge, publish and let you know next week.

TheZoq2 commented:
Made all the links open traces and a small quick start window. I also updated the date to monday 27th and I think you have push access to this branch if you want to adjust it to when you actually release it
